bnd-shared-maven-lib/.classpath

```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
I thought we had discussed that bnd-shared-maven-lib would remain part of the maven build, but some of the classes would be moved down to another project where the classes could be conditional packaged in m2e plugins and maven-dependency-plugin included in bnd-shared-maven-lib.

Reply:
The naming suggests a maven lib but you are building in a bundle which one could assume will be installed in Eclipse with the Bndtools.m2e feature. But instead you are conditional packaging it into the bndtools m2e bundle. So why make a bundle that is never used as a bundle? bndlib _is_ used as a bundle (and a jar).

Reply:
Also this change converts bnd-shared-maven-lib to no longer have bnd-plugin-parent as its parent. Not sure if this is a real issue. But it is a change that is observable.

Reply:
The fact that bnd-shared-maven-lib is a bundle (now) is just a side effect of being built by bnd workspace. It is not the intention to use it that way (this is true of many projects inside the bnd repo, not sure how this would be any different.)

Reply:
In order for bnd-shared-maven-lib to be used by the maven plugins, it must be released to maven central. That means you are making a bundle available to the world. That it, it becomes API. Currently, it is essentially impl detail to the maven plugins which I think it should remain.

Reply:
I'm still not sure I understand. How is it any different than what we have today?

https://search.maven.org/artifact/biz.aQute.bnd/bnd-shared-maven-lib/4.1.0/jar

Reply:
That is a jar with a pom. This PR makes it a bundle with exported packages (API) which, of course, should be baselined since it is API. I don't think this jar is API. It is shared impl detail because we cannot easily do conditionalpackage in a maven build. I will note you clearly see this code as impl detail since you conditionalpackage it into the m2e bundle vs. importing the package and adding the bundle to the m2e feature.

Reply:
I'm not sure how to address this particular issue.

Reply:
My suggestion is to leave bnd-shared-maven-lib project in the maven build. Move any packages that must be shared to bndtools.m2e project (so bndtools.m2e has use of the packages). Then change bnd-shared-maven-lib to pull those packages into it using maven-dependency-plugin on the bndtools.m2e jar which will be in the local repo.

The we have the shared package source code in one place (bndtools.m2e project) and copies of the shared package class files in 2 places (bndtools.m2e jar and bnd-shared-maven-lib jar).

Reply:
If i do that, there will be zero code in the `bnd-shared-maven-lib` project!